pkg/querier/block_store.go

```diff
@@ -268,14 +217,15 @@ func (u *UserStore) getOrCreateStore(userID string) (*store.BucketStore, error)
 		userBkt,
 		filepath.Join(u.cfg.BucketStore.SyncDir, userID), // The fetcher stores cached metas in the "meta-syncer/" sub directory
 		fetcherReg,
-		[]block.MetadataFilter{
-			// List of filters to apply (order matters).
+		// The input filters MUST be before the ones we create here (order matters).
+		append(u.filters, []block.MetadataFilter{
 			block.NewConsistencyDelayMetaFilter(userLogger, u.cfg.BucketStore.ConsistencyDelay, fetcherReg),
 			block.NewIgnoreDeletionMarkFilter(userLogger, userBkt, u.cfg.BucketStore.IgnoreDeletionMarksDelay),
 			// Filters out duplicate blocks that can be formed from two or more overlapping
 			// blocks that fully submatches the source blocks of the older blocks.
+			// TODO(pracucci) can this cause troubles with the upcoming blocks sharding in the store-gateway?
```

Comment:
I don't think it will. I think deduplicate filter removes old blocks if new blocks cover it completely, but we better verify it.

Reply:
I think we may have troubles with the consistency check, at least as I did design it. I will write down different scenarios in tests and let's see (once we'll reach the point to build the consistency check).